dll/win32/msgina/shutdown.c

```diff
@@ -1059,45 +1279,22 @@ ShutdownDialogProc(
                 if (!pContext->bCloseDlg)
                 {
                     pContext->bCloseDlg = TRUE;
-
-                    if (pContext->bIsDialogModal)
-                    {
-                        EndDialog(hDlg, 0);
-                    }
-                    else
-                    {
-                        DestroyWindow(hDlg);
-                        PostQuitMessage(0);
-                    }
+                    EndDialog(hDlg, 0);
```

Comment:
Same question/remark as in shell32:
Perhaps `EndDialog(hwnd, IDCANCEL);` ? (while it's true that here, nobody clicked the `IDCANCEL` button, the dialog effectively quits as if a cancel action occurred.)